## Background

PetMind is a retailer of products for pets. They are based in the United States.
PetMind sells products that are a mix of luxury items and everyday items. Luxury items include toys. Everyday items include food.
The company wants to increase sales by selling more everyday products repeatedly. They have been testing this approach for the last year.
They now want a report on how repeat purchases impact sales.

## Dataset

The dataset contains the sales records in the stores last year.

In [1]:
# import libraries
import numpy as np
import pandas as pd

# load dataset
df = pd.read_csv('Dataset/pet_supplies_2212.csv')

In [2]:
df.head()

product_id  category animal    size  price    sales  rating   
0           1      Food   Bird   large   51.1  1860.62     7.0  \
1           2   Housing   Bird  MEDIUM  35.98   963.60     6.0   
2           3      Food    Dog  medium  31.23   898.30     5.0   
3           4  Medicine    Cat   small  24.95   982.15     6.0   
4           5   Housing    Cat   Small  26.18   832.63     7.0   

   repeat_purchase  
0                1  
1                0  
2                1  
3                1  
4                1

In [3]:
n_rows = df.shape[0]
n_columns = df.shape[1]

print(f'Dataset has {n_rows} rows and {n_columns} columns.')

Dataset has 1500 rows and 8 columns.


## Data Cleaning with Python

### Check Missing values and data format

#### Check data type of dataframe

In [4]:
print(df.dtypes)

product_id           int64
category            object
animal              object
size                object
price               object
sales              float64
rating             float64
repeat_purchase      int64
dtype: object


#### Check is there any missing values in dataframe

In [5]:
print(df.isna().sum())

product_id           0
category             0
animal               0
size                 0
price                0
sales                0
rating             150
repeat_purchase      0
dtype: int64


#### Check each column data criteria

In [6]:
columns = ['category', 'animal', 'size', 'price', 'sales', 'rating']

for col in columns:
    print(f'{col} column -')
    print(df[col].value_counts())
    print('\n\n')


category column -
category
Equipment    370
Food         260
Toys         255
Medicine     237
Housing      227
Accessory    126
-             25
Name: count, dtype: int64



animal column -
animal
Cat     567
Fish    369
Dog     367
Bird    197
Name: count, dtype: int64



size column -
size
small     372
medium    233
Small     225
SMALL     157
Medium    150
large     122
MEDIUM    109
Large      75
LARGE      57
Name: count, dtype: int64



price column -
price
unlisted    150
22.98        13
28           10
28.03         8
19.94         8
           ... 
38.66         1
36            1
19.97         1
39.19         1
25.79         1
Name: count, Length: 707, dtype: int64



sales column -
sales
1034.36    3
742.87     2
1342.90    2
1031.11    2
1348.27    2
          ..
1072.20    1
727.87     1
1460.29    1
1196.56    1
826.08     1
Name: count, Length: 1476, dtype: int64



rating column -
rating
5.0    304
6.0    299
4.0    283
3.0    190
7.0    143
8.0     61
2.0     43
9.0  

There are:
- 25 missing values in the category column (as '-' ).
- 150 missing values in the price column (as 'unlisted' ).
- 150 missing values in the rating column (as 'NA' ).




In this overview, we can see that there are several columns that should have missing values but have initial values that are not appropriate, such as:

- category, in the form of '-'
- price, in the form of 'unlisted'
- size, in the form of many combination of upper and lower cases for same size
- rating, in the form of 'NA'



### Fill missing values

- Missing values in the category column must be changed to 'Unknown',
- Missing values in the price column must be changed to the median value of price,
- Missing values in the rating column must be changed to '0'.

##### Category Column

In [7]:
# replace hyphens in 'category' column with 'unknown'
df['category'] = df['category'].str.replace('-', 'unknown')

##### Price Column

In [8]:
# replace "unlisted" values in "price" column with the median of the column
median_price = pd.to_numeric(df['price'], errors='coerce').median()  # calculate median price
df['price'] = pd.to_numeric(df['price'], errors='coerce').replace(np.nan, median_price)  # replace "unlisted" with median

##### Rating Column

In [9]:
df['rating'] = df['rating'].fillna(0)

In [10]:
df.isna().any()

product_id         False
category           False
animal             False
size               False
price              False
sales              False
rating             False
repeat_purchase    False
dtype: bool

### Data Cleaning

##### Size Column

In [11]:
# convert all values in "size" column to lowercase
df['size'] = df['size'].str.lower()

##### repeat_purchase Column

In [12]:
def repeat_purchase(value):
    if value == 1:
        return 'Y'
    return 'N'

df.rename(columns={"repeat_purchase": "rp"}, inplace=True)
df['repeat_purchase'] = df['rp'].apply(repeat_purchase)


df.drop(['rp'], axis=1, inplace=True)

In [13]:
df.head()

product_id  category animal    size  price    sales  rating repeat_purchase
0           1      Food   Bird   large  51.10  1860.62     7.0               Y
1           2   Housing   Bird  medium  35.98   963.60     6.0               N
2           3      Food    Dog  medium  31.23   898.30     5.0               Y
3           4  Medicine    Cat   small  24.95   982.15     6.0               Y
4           5   Housing    Cat   small  26.18   832.63     7.0               Y

Save the clean dataframe in a new **_csv_** file

In [14]:
df.to_csv('Dataset/clean_pet_df.csv')

---

## Tableau Dashboards' Link: 

https://public.tableau.com/shared/24NS2TYRD?:display_count=n&:origin=viz_share_link

## Visualization to show how many products are repeat purchases


<img src="Analysis Images/PetMind Sales Report-2.png" alt="PetMind Sales Report-2" />


It can be seen that the difference between repeatable purchase and non-repeatable purchase is as much as around 300 purchases. 

From the chart, we see that the equipment category have most repeated puchases compare than others. However, when compared as a whole, the Equipment category and the Accessory category differ significantly.

Moreover, in PetMind, items for cats are the most often bought items. However, given the limitations of our data, it is still unclear whether or whether this outcome is due to cats being more accessible to products than other animals.


<img src="Analysis Images/PetMind Sales Report-3.png" alt="PetMind Sales Report-3" />


Now we can see it more clearly that the majority of purchases of goods for cats are made in the category of equipment, although it actual purchases in other categories are still above average. The treemap also reveals that each animal has a distinct market under each category, with cats falling under the category of equipment, dogs under housing, fish under medicine, and birds under food.

## Distribution of all sales


<img src="Analysis Images/PetMind Sales Report-4.png" alt="PetMind Sales Report-4" />


Overall, it is clear that products with repeated purchases have significantly higher total sales than those without. 

Equipment products have the largest total sales when broken down by category, while the ranking of other categories has changed. The same phenomenon occurs if we divide it up according to animal, with products for cats having the highest total sales. This occurrence may be an indication that a large number of transactions in one category or animal does not guarantee a high volume of sales overall.

## Relationship between repeat purchases and sales


<img src="Analysis Images/PetMind Sales Report-5.png" alt="PetMind Sales Report-5" />


From that graph, we can draw the conclusion that repeat purchases have a positive impact on overall sales because the sum of sales for repeat purchases, broken down by category and animal type, are all larger than those for non-repeat purchases. Through this outcome, PetMind can also utilize it as an approach to raise the likelihood that customers would make additional purchases, which will boost sales outcomes. Additionally, some of the preceding graphs can be used as a guide by PetMind when deciding which animal product categories or subcategories should be preserved or developed in order to maximize profit.


## **Conclusions**

Products with the Equipment category have the most number of observations and sales.
There should be adjustments made about the products PetMind should concentrate on because the amount of observations and sales in each category are not spread equally.
PetMind can take use of this finding to raise the likelihood that customers will make repeat purchases since repeat purchases have a favorable correlation with overall sales.